Posh-ACME/Plugins/EfficientIP.ps1

```diff
@@ -0,0 +1,237 @@
+function Get-CurrentPluginType { 'dns-01' }
+
+function Add-DnsTxt {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory,Position=0)]
+        [string]$RecordName,
+        [Parameter(Mandatory,Position=1)]
+        [string]$TxtValue,
+        [Parameter(Mandatory,Position=2)]
+        [string]$EIPUsername,
+        [Parameter(Mandatory,Position=3)]
+        [securestring]$EIPPassword,
+        [Parameter(Mandatory,Position=4)]
+        [string]$EIPHostname,
+        [Parameter(Mandatory,Position=5)]
+        [string]$EIPDNSName,
+        [Parameter(Mandatory,Position=6)]
+        [string]$EIPView,
+        [Parameter(ValueFromRemainingArguments)]
+        $ExtraParams
+    )
+
+    # convert the secure secret to a normal string
+    $EIPPasswordInsecure = [pscredential]::new('a',$EIPPassword).GetNetworkCredential().Password
+
+    $Endpoint = "dns_rr_add"
+    $queryParams += @{rr_name = $RecordName}
+    $queryParams += @{value1 = $TxtValue}
+    $queryParams += @{rr_type = "TXT"}
+    $queryParams += @{rr_ttl = "1"}
+
+    $queryString = ($queryParams.GetEnumerator() | ForEach-Object { "$($_.Key)=$($_.Value)" }) -join '&'
+    $Parameters = "dnsview_name=$($EIPView)&add_flag=new_only&check_value=yes&dns_name=$($EIPDNSName)&$($queryString)"
```

Comment:
The view is optional - If not specified the dnsview_name should not be included in the request